src/lib/helpers/formErrorHandler.ts

```
@@ -0,0 +1,56 @@
+		serverErrorMessage = 'An error occurred. Please try again.'
+	} = options;
+
+	return () => {
+		// set loading state immediately when form submission starts
+		setLoading?.(true);
+		setError?.(null);
+
+		return async ({ result, update }: { result: ActionResult; update: () => Promise<void> }) => {
+			if (result.type === 'failure') {
+				// server returned an error/failure
+				const errorMsg = (result.data as any)?.error || serverErrorMessage;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)